src/Features/LanguageServer/Protocol/Handler/Diagnostics/DocumentPullDiagnosticHandler.cs

```diff
@@ -91,16 +83,32 @@ internal static ImmutableArray<Document> GetRequestedDocument(RequestContext con
             if (context.Document == null)
             {
                 context.TraceInformation("Ignoring diagnostics request because no document was provided");
-                return ImmutableArray<Document>.Empty;
+                return ImmutableArray<IDiagnosticSource>.Empty;
             }
 
             if (!context.IsTracking(context.Document.GetURI()))
             {
                 context.TraceWarning($"Ignoring diagnostics request for untracked document: {context.Document.GetURI()}");
-                return ImmutableArray<Document>.Empty;
+                return ImmutableArray<IDiagnosticSource>.Empty;
+            }
+
+            return ImmutableArray.Create<IDiagnosticSource>(new DocumentDiagnosticSource(context.Document));
+        }
+
+        private record struct DocumentDiagnosticSource(Document Document) : IDiagnosticSource
```

Comment:
nit: put in separate file.

Reply:
normally I would agree - however this is a private type used only by the class its inside, and it is small enough I'm not sure its worth a partial.  Unless theres a strong reason to move to its own file I definitely prefer it here.

Reply:
wfm.